src/lib/parser/statement/splitter.rs

```diff
@@ -109,19 +120,40 @@ impl<'a> StatementSplitter {
             math_paren_level: 0,
         }
     }
+
+    fn get_statement(&mut self, character: u8) -> StatementVariant<'a> {
+        let variant = if self.flags.contains(Flags::AND) {
+                self.flags.remove(Flags::AND);
+                self.data = &self.data[self.read + 1..].trim();
+                StatementVariant::And(&self.data)
+            } else if self.flags.contains(Flags::OR) {
+                self.flags.remove(Flags::OR);
+                self.data = &self.data[self.read + 1..].trim();
+                StatementVariant::Or(&self.data)
+            } else {
+                if character == b'&' {
+                    self.flags.insert(Flags::AND);
+                }
+                if character == b'|' {
+                    self.flags.insert(Flags::OR);
+                }
```

Comment:
These two if conditions can be removed, and replaced with

```rust
self.flags |= new_flag;
```

Reply:
will do